### ISSUE TITLE
[QUESTION] Yup validation of multi level object

### ISSUE THREAD

**Issue by nissane** [jquense/yup#651]
I am using Yup with formik.  My values look like this:

```
  "values": {
    "startTime": "2019-10-13T01:04:11.596Z",
    "endTime": null,
    "phraseType": {
      "id": "HELLO"
    }

  phraseValue: Yup.number().positive().when('phraseType', {
    is: (val) => val.id === 'HELLO',
    then: Yup.string().required('required'),
    otherwise: Yup.string().notRequired()
  }),
```

It doesn't seem I am able to access the id off of the val object.  Val is being returned as a string "[Object][Object]" instead of the actual object.  How do I use when() with a key to get back an object?

Thanks!

**Comment by jquense** [comment]
Your path looks wrong, it'd be `values.phraseType`

**Comment by nissane** [comment]
No, that's not the issue I am able to access other properties without prefacing with values.  I am using this with formik, so that is not needed. 
I'll rephrase it without values:
```
  {
    "startTime": "2019-10-13T01:04:11.596Z",
    "endTime": null,
    "phraseType": {
      "id": "HELLO"
    }

  phraseValue: Yup.number().positive().when('phraseType', {
    is: (val) => val.id === 'HELLO',
    then: Yup.string().required('required'),
    otherwise: Yup.string().notRequired()
  }),
```
The issue is that you cannot (as far as I can tell) access properties on an object using when->is.  It returns a string of "[Object][Object]" instead of returning the object.

**Comment by nissane** [comment]
Furthermore, doing .when('values.phraseType') returns undefined.  Can you please reopen?  Many Thanks.

**Comment by jquense** [comment]
can you provide a runnable reproduction? I'll take a look

**Comment by nissane** [comment]
Thank you.  Does this work for you?  

https://runkit.com/sberic/yup-examples

**Comment by nissane** [comment]
Can you reopen this so it doesn't get lost? Really appreciate it!

**Comment by jquense** [comment]
```
const yup = require('yup');

var contactSchema = yup.object().shape({
  name: yup.string().when('phraseType', {
    is: (val) => val.id === 3,
    then: yup.string().required(),
  }),

});


let contact = {
    "phraseType": {
      "id": 3
    }
}

console.log(contactSchema.validateSync(contact))
```

this works fine?

**Comment by nissane** [comment]
I don't think so, I created an example with pretty much that. You can see it throws an undefined exception.
https://runkit.com/sberic/yup-examples

**Comment by jquense** [comment]
that example is using a very old version of yup I ran the example I posted above and it does work with the current version at least

**Comment by jquense** [comment]
if you're still stuck the reason your example is throwing is because `phaseType` is not defined in your schema.

**Comment by nissane** [comment]
ok thanks, I got everything up to the latest version.  I found my problem.  In my code I also had:
 phraseType: yup.string().required(),
that needed to be phraseType: yup.object().required().

I updated my example if your interested.  if you use the line:
 phraseType: yup.string().required(),

the validation fails

if you change it to:
 phraseType: yup.object().required(),

validation works. Not a bug, just letting you know. really appreciate you researching.
